ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py

```diff
@@ -1611,3 +1611,36 @@ def test_tucker_tensorly(tol_norm_2, tol_max_abs, shape, ranks):
         ivy.max(ivy.abs(rec_svd - rec_random)) < tol_max_abs,
         "abs norm of difference between svd and random init too high",
     )
+
+
+@handle_test(
+    fn_tree="functional.ivy.experimental.general_inner_product",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
+        min_num_dims=2,
+        max_num_dims=3,
+        min_dim_size=2,
+        max_dim_size=5,
+        min_value=1,
+        max_value=10.0,
+        num_arrays=2,
+        shared_dtype=True,
+    ),
+    n_modes=st.integers(min_value=0),
```

Comment:
can also be `None` and you should put an upper bound on it : `max_value = min(dims(a), dims(b))`